### PR TITLE
Dtsstci 120 - Audit Unit Tests in Case API - Notification Module

### PR DESCRIPTION
**Pull request by Jefex7**
### Change description ###

Audit Unit Tests in Case API - Notification Module

### JIRA link (if applicable) ###

[https://tools.hmcts.net/jira/browse/DTSSTCI-120](url)

**Before merging a pull request make sure that:**

- [X] tests have been updated / new tests has been added (if needed)
- [X] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/notification/NotificationHelper.java [hunk 1]
```diff
@@ -182,14 +182,14 @@ public void setRecordingTemplateVars(Map<String, Object> templateVars, Listing l
     }
 
     public Map<String, String> buildDocumentList(DynamicMultiSelectList documentList, int docAttachLimit) {
-        Map<String, String> uploadedDocuments = new HashMap<>();
+        final Map<String, String> uploadedDocuments = new HashMap<>();
 
         int count = 0;
         if (!ObjectUtils.isEmpty(documentList.getValue()) && documentList.getValue().size() > 0) {
```
Comment (oleon23) [inline src/main/java/uk/gov/hmcts/sptribs/notification/NotificationHelper.java:188]: You can replace this condition with `ObjectUtils.isNotEmpty(documentList.getValue())`

Comment (oleon23) [inline src/main/java/uk/gov/hmcts/sptribs/notification/NotificationHelper.java:188]: Also can add tests for when document list is empty or null in `NotificationHelperTest`?

File: src/main/java/uk/gov/hmcts/sptribs/notification/NotificationHelper.java [hunk 2]
```diff
@@ -210,26 +210,27 @@ public Map<String, String> buildDocumentList(DynamicMultiSelectList documentList
...
-        return null != listing.getHearingFormat() && listing.getHearingFormat().equals(HearingFormat.VIDEO);
+        return listing.getHearingFormat() != null && listing.getHearingFormat().equals(HearingFormat.VIDEO);
     }
 
     private boolean isTelephoneFormat(Listing listing) {
-        return null != listing.getHearingFormat() && listing.getHearingFormat().equals(HearingFormat.TELEPHONE);
+        return listing.getHearingFormat() != null && listing.getHearingFormat().equals(HearingFormat.TELEPHONE);
     }
 
     public void addHearingPostponedTemplateVars(CicCase cicCase, Map<String, Object> templateVars) {
```
Comment (oleon23) [inline src/main/java/uk/gov/hmcts/sptribs/notification/NotificationHelper.java:224]: Can add a test case for the `selectedHearingDateTime` to be null and assert the correct behaviour

File: src/test/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCICTest.java [hunk 3]
```diff
@@ -255,50 +253,94 @@ void shouldThrowNotificationExceptionWhenClientFailsToSendEmail()
             any());
     }
 
-    @Ignore
-    void shouldThrowNotificationExceptionWhileFileUploadToSendEmail()
-        throws IOException {
+    @Test
+    void shouldThrowIOExceptionWhenClientFailsToSendEmail()
```
Comment (gmmagruder) [inline src/test/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCICTest.java:257]: Please also add a test that covers the condition when an IOException is thrown

Comment (tomxelliott) [reply src/test/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCICTest.java:257]: As per Gabrielle's comment this is not testing IOException - it is testing NotificationException. Please change the name and add a test case for IOException.

File: src/test/java/uk/gov/hmcts/sptribs/common/notification/NotificationHelperTest.java [hunk 4]
```diff
@@ -57,6 +59,9 @@ public class NotificationHelperTest {
     @InjectMocks
     private NotificationHelper notificationHelper;
 
+    @Mock
+    private CicCase cicCase;
+
     @Test
     void setRecordingTemplateVarsTest() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/common/notification/NotificationHelperTest.java:66]: I would suggest the following test cases for the method setRecordingTemplateVars. (you can modify/remove the existing ones accordingly)
1. A test case which passes all the if conditions in that method
2. A test case which fails in all the if conditions in that method (selected venue not null)
3. A test case for CIC_CASE_HEARING_VENUE with HearingVenueNameAndAddress not null (not video or telephone format and selected venue null)
4. A test case for CIC_CASE_HEARING_VENUE with HearingVenueNameAndAddress null, not video or telephone format and selected venue null
5. A test case for CIC_CASE_HEARING_VENUE with video format, but not telephone format
6. A test case for CIC_CASE_HEARING_VENUE with telephone format, but not video format

File: src/test/java/uk/gov/hmcts/sptribs/common/notification/NotificationHelperTest.java [hunk 5]
```diff
@@ -293,6 +298,18 @@ void shouldAddHearingPostponedTemplateVars() {
         assertThat(templateVars.get(HEARING_TIME)).isEqualTo("11:00");
     }
 
+    @Test
+    void shouldAddHearingPostponedTemplateVarsWithNull() {
```
Comment (rajaram-nagarajan) [inline src/test/java/uk/gov/hmcts/sptribs/common/notification/NotificationHelperTest.java:302]: A new test case needs to be added with invalid input (wrong format) for selected hearing date and time.

File: src/main/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCIC.java [hunk 6]
```diff
@@ -36,30 +36,44 @@
...
-    private NotificationClient notificationClient;
+    private final NotificationClient notificationClient;
 
     @Autowired
-    private EmailTemplatesConfigCIC emailTemplatesConfig;
+    private final EmailTemplatesConfigCIC emailTemplatesConfig;
 
     @Autowired
-    private IdamService idamService;
+    private final IdamService idamService;
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCIC.java:46]: We have another notification module under common module. If we don't have any JIRA ticket to cover that, feel free to raise one.

**Review by rajaram-nagarajan — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCIC.java (hunk 6 above)
Comment (Jefex7) [reply src/main/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCIC.java:46]: Maybe a task can be added to this ticket to remove the common > notification package and place everything inside the notification module ?
[https://tools.hmcts.net/jira/browse/DTSSTCI-318](url)

File: src/test/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCICTest.java (hunk 3 above)
Comment (Jefex7) [reply src/test/java/uk/gov/hmcts/sptribs/notification/NotificationServiceCICTest.java]: Apologies, I changed the name of the method but it seems that the implementation of the method sendEmail throws the IOException wrapped in a notificationException. So I haven't been able to find a way to just test IOException without the notificationException being present as well, maybe I am missing something.

**Review by tomxelliott — APPROVED** [review]
